### DISCUSSION TITLE
How to install more than one GTM in same application

### DISCUSSION THREAD

**Discussion by medeirosrafael** [gtm-support/vue-gtm#161] [Q&A]
I need multiple GTM on the same website.
The first one (XXX) will be active in all routes.
The second one (YYY) will be active only in the route "/b".
And the third one (ZZZ) will be active in the route "/c".

It's that possible at all?

**Comment by medeirosrafael** [comment]
If i'm not wrong, multiple instances will share same dataLayer.
So, even if it were possible to ignore views in specific ID, all my instances share the same dataLayer, so, all my instances will keep tracking events..
And all events will be duplicated (or triplicated) in both GTM..

Someone can give me a light?

**Reply by Shinigami92** [reply]
As far as I know, yes, there is only one `dataLayer` and google is accessing exactly this window property.
So I assume you need to add somehow properties to your events and then filter on them.
Sadly I don't know anything about that cause I just took this repo over to maintain the code, because my previous company used it. But now I don't use it myself anymore cause my current company's software does not use GTM.
But I can say that I'm active and can help to fix bugs and add features when you explain what and why I have to add something.

**Reply by medeirosrafael** [reply]
Thanks for replying.

My need is load some GTM ID's only in specifc vue routes.. 

I only can see it as possible if I use together ignoredViews (per ID not globaly as it's today) and custom dataLayer property name (https://developers.google.com/tag-platform/tag-manager/web/datalayer#rename_the_data_layer)..

So, maybe it'll necessary change this project and also the core project ..

But i'm open to new ideas

**Reply by Shinigami92** [reply]
Would you like to propose a PR? Then I can look into your idea/code and make suggestions on that
